docs/whichkey/reference.md

```diff
@@ -0,0 +1,281 @@
+See [Extra](extra#use-non-character-keys) page for detail on usage.
+:::
+
+:::note
+
+#### Pass-in when clause
+
+Since vscode doesn't allows extension to read context key-value ([vscode#10471](https://github.com/microsoft/vscode/issues/10471)),
+we will have to relied on `when` clause evaluation in vscode shortcut to pass the context to which-key.
+See [conditional bindings](./extra.md#when) for more details on the usage.
+:::
+
+### Describe Bindings
+
+Command: `whichkey.describeBindings`
+
+Description:
+This is similar to [helm-descbinds](https://github.com/emacs-helm/helm-descbinds) which can search and execute the binding ([vscode-which-key#12](https://github.com/VSpaceCode/vscode-which-key/issues/12)).
+Note that it will only shows described bindings menu for the currently displayed bindings (and its sub-bindings).
+
+### Show Transient
+
+Command: `whichkey.showTransient`
+
+Args: [`TransientMenuConfig`](#transientmenuconfig) | `string`
+
+Description: Show a transient with the definition in the argument. The definition can be either
+
+- object of [`TransientMenuConfig`](#transientmenuconfig)
+- a `string` referencing the location of in the configuration to get `TransientMenuConfig`
+
+### Repeat Recent
+
+Command: `whichkey.repeatRecent`
+
+Description:
+A command to show a which-key menu with key 1 to 9 to repeat the most recently executed command bindings.
+
+### Recent Most Recent
+
+Command: `whichkey.repeatMostRecent`
+
+Description:
+A command to repeat the command binding most recently executed on which-key.
+
+### Toggle Zen Mode
+
+Command: `whichkey.toggleZenMode`
+
+Description:
+A command to toggle zen mode for transient menu, which will show/hide all the menu items to save screen real estate.
+
+### Open File
+
+Command: `whichkey.openFile`
+
+Description:
+This is a command to get around issue where vscode doesn't provide a single cross platform command to open file.
+See [vscode-which-key#26](https://github.com/VSpaceCode/vscode-which-key/issues/26).
+
+## Context
+
+The extension will also set the following context when applicable so it can be used in `when` clause of vscode shortcuts.
+
+`whichkeyActive` is a boolean that will be set to `true` when whichkey is active including the time when it is excluding a command.
+This is rarely used and most of the use case can be replaced with `whichKeyVisible`
+
+`whichkeyVisible` is a boolean that will be set to `true` when whichkey menu is visible. Note that this will be `false` when transient menu is visible.
+
+`transientVisible` is a boolean that wil lbe set to `true` when the transient menu is visible.
+
+## Config
+
+The extension uses the following configurations.
+
+### Delay
+
+Key: `whichkey.delay`
+
+Type: `number`
+
+Description:
+Delay (in milliseconds) for which-key menu to display. Setting this to positive value will delay the display of the which-key menu items while
```

Comment:
```suggestion
Delay (in milliseconds) for the which-key menu to be displayed after it has been triggered.
Setting this to positive value will just delay the display of the which-key menu items, while
```

Reply:
```suggestion
Delay (in milliseconds) for the which-key menu items to be displayed.
Setting this to positive value will just delay the display of the which-key menu items, while
```
I feel like add "it has been triggered" might add more confusion, especially I was not sure what "it" refers to and the word trigger doesn't seem right here. I also thought different phrase like "after the show command is executed". But, it seems like it not adding value while being a long sentence. So I kept it succinct and short in the suggestion.